sources/slack/helpers.py

```diff
@@ -0,0 +1,206 @@
+        """
+        params = params or {}
+        params["limit"] = self.page_size
+        if next_cursor:
+            params["cursor"] = next_cursor
+        return params
+
+    def url(self, resource: str) -> str:
+        """
+        Generate the URL to use for the request.
+
+        Args:
+            resource: The resource to get pages for (e.g. conversations.list).
+        """
+        return urljoin(SLACK_API_URL, resource)
+
+    def _get_next_cursor(self, response: Dict[str, Any]) -> Any:
+        """
+        Get the next cursor from the response.
+
+        Args:
+            response: The response from the Slack API.
+        """
+        cursor_jsonpath = "$.response_metadata.next_cursor"
+        return next(extract_jsonpath(cursor_jsonpath, response), None)
+
+    def _convert_datetime_fields(
+        self, item: Dict[str, Any], datetime_fields: List[str]
+    ) -> Dict[str, Any]:
+        """Convert timestamp fields in the item to pendulum datetime objects.
+
+        The item is modified in place.
+
+        Args:
+            item: The item to convert
+            datetime_fields: List of fields to convert to pendulum datetime objects.
+
+        Returns:
+            The same data item (for convenience)
+        """
+        if not datetime_fields:
+            return item
+
+        for field in datetime_fields:
+            if timestamp := next(extract_jsonpath(field, item), None):
+                if isinstance(timestamp, str):
+                    timestamp = float(timestamp)
+                if timestamp > 1e10:
+                    timestamp = timestamp / 1000
+                pendulum_dt = pendulum.from_timestamp(timestamp)
+                item = update_jsonpath(field, item, pendulum_dt)
+        return item
+
+    def get_pages(
+        self,
+        resource: str,
+        response_path: str = None,
+        params: Dict[str, Any] = None,
+        datetime_fields: List[str] = None,
+        context: Dict[str, Any] = None,
+    ) -> Iterable[TDataItem]:
+        """Get all pages from slack using requests.
+        Iterates through all pages and yield each page items.\
+
+        Args:
+            resource: The resource to get pages for (e.g. conversations.list).
+            response_path: The path to the list of items in the response JSON.
+            params: Query params to include in the request.
+            datetime_fields: List of fields to convert to pendulum datetime objects.
+            context: Additional context to add to each item.
+
+        Yields:
+            List of data items from the page
+        """
+        has_next_page = True
+        next_cursor = None
+
+        # Iterate through all pages
+        while has_next_page:
+            # Make the request
+            response = requests.get(
+                url=self.url(resource),
+                headers=self.headers,
+                params=self.parameters(params or {}, next_cursor),
+                timeout=60,
```

Comment:
please remove timeout. our requests have it and it can be also configured by the user